src/web/releases.rs

```diff
@@ -244,64 +252,106 @@ fn get_releases_by_owner(
+) -> (i64, Vec<Release>) {
+    let query = query.trim().to_lowercase();
     let offset = (page - 1) * limit;
-    let mut packages = Vec::new();
 
-    let rows = match conn.query(
-        "SELECT crates.name,
-                                    releases.version,
-                                    releases.description,
-                                    releases.target_name,
-                                    releases.release_time,
-                                    releases.rustdoc_status,
-                                    ts_rank_cd(crates.content, to_tsquery($1)) AS rank
-                                 FROM crates
-                                 INNER JOIN releases ON crates.latest_version_id = releases.id
-                                 WHERE crates.name LIKE concat('%', $1, '%')
-                                    OR crates.content @@ to_tsquery($1)
-                                 ORDER BY crates.name = $1 DESC,
-                                    crates.name LIKE concat('%', $1, '%') DESC,
-                                    rank DESC
-                                 LIMIT $2 OFFSET $3",
-        &[&query, &limit, &offset],
-    ) {
-        Ok(r) => r,
-        Err(_) => return None,
+    let statement = "
+        SELECT
+            crates.name,
+            releases.version,
+            releases.description,
+            releases.target_name,
+            releases.release_time,
+            releases.rustdoc_status,
+            crates.github_stars,
+            -- Get the total number of results, disregarding the limit 
+            COUNT(*) OVER() as total
+        FROM releases
+        INNER JOIN crates on releases.crate_id = crates.id
+        WHERE
+            -- Only select the newest release by release time
+            releases.id = (
+                SELECT releases.id
+                FROM releases
+                -- Filter unbuilt/failing builds and yanked releases
+                WHERE
+                    releases.crate_id = crates.id
+                    AND releases.rustdoc_status
+                    AND NOT releases.yanked
+                ORDER BY releases.release_time DESC
+                LIMIT 1
+            )
+            -- Only select releases/crates that pass our criteria:
+            --      - Levenshtein distance between the name and query is acceptable
+            --      - The query sandwiched between wildcards matches the crate's name
+            --      - The query matches the release's description
+            AND (
+                -- Turn the levenshtein distance into a percentage using `distance / max(query.len(), crates.name.len())`
+                -- this percentage is normalized and allows us to imperatively compare the 'sameness' of different names
+                (((CAST(GREATEST(char_length($1), char_length(crates.name)) AS DOUBLE PRECISION) - CAST(levenshtein(CAST(crates.name AS TEXT), CAST($1 AS TEXT)) AS DOUBLE PRECISION))
```

Comment:
As discussed on discord, the casts make this hard to read and the logic is pretty confusing. I would rather do something like `levenshtein(crates.name, $1) < ROUND(LOG(2, char_length($1)))`.

Some sample cutoffs:

| input_length | cutoff |
|--------------|--------|
| 2            | 1      |
| 5            | 2      |
| 10           | 3      |
| 25           | 5      |